Obsolete warning for authenticator class.

## Expected Behavior
Implementing custom authenticator should not result error:
`warning CS0618: 'Parameter' is obsolete: 'Use Add[XXX]Parameter methods of IRestRequest instead of instantiating the Parameter class.'`

## Actual Behavior
When implementing custom authenticator based on Docs and actual build in code from:
https://github.com/restsharp/RestSharp/tree/dev/src/RestSharp/Authenticators
https://restsharp.dev/usage/authenticators.html#custom-authenticator

I'm receiving an error:
`warning CS0618: 'Parameter' is obsolete: 'Use Add[XXX]Parameter methods of IRestRequest instead of instantiating the Parameter class.'`

The `AuthenticatorBase` class requires override 
```c#
protected abstract Parameter GetAuthenticationParameter(string accessToken);
```
method

and does not allow override 

```c#
public void Authenticate(IRestClient client, IRestRequest request)
```

method.

The build in authenticators are using forbidden statement.

To be consistent all authenticator code should be removed and and moved to AddXXXAuthentication methods.

--- a/src/RestSharp/Parameter.cs
+++ b/src/RestSharp/Parameter.cs
@@ -24,7 +24,6 @@ namespace RestSharp {
     /// <summary>
     /// Parameter container for REST requests
     /// </summary>
-    [Obsolete("Use Add[XXX]Parameter methods of IRestRequest instead of instantiating the Parameter class.")]
     public class Parameter : IEquatable<Parameter> {
         public Parameter(string name, object? value, ParameterType type, bool encode = true) {
             if (type != ParameterType.RequestBody)

